src/wallet/models.py

```diff
@@ -46,8 +49,31 @@ def by_direction(self, direction):
         else:
             return self.incoming()
 
+    def _latest_stamps(self):
+        # Filter out the latest status timestamps for every transaction
+        stamps = self.annotate(latest_stamp=models.Max('states__date_created'))
+        return stamps.values_list('latest_stamp', flat=True)
+
+    def by_status(self, status=None):
```

Comment:
Or wait, both conditions have to hold at the same time. If this should work, we have to exclude all `states` except the latest in the first query.